llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1301,9 +1303,26 @@ static std::optional<Instruction *> instCombineSVEAllActive(IntrinsicInst &II,
   return &II;
 }
 
+// Optimize operations that take an all false predicate or send them for
+// canonicalization.
+static std::optional<Instruction *>
+instCombineSVEAllOrNoActive(InstCombiner &IC, IntrinsicInst &II,
+                            Intrinsic::ID IID) {
+  if (match(II.getOperand(0), m_ZeroInt())) {
+    if (II.getIntrinsicID() != IID)
+      return IC.replaceInstUsesWith(II, II.getOperand(1));
+    else
+      return IC.replaceInstUsesWith(II, UndefValue::get(II.getType()));
```

Comment:
nit: maybe add a comments
// llvm_ir_u, pred(0), op1, op2   - Spec says to return undef when all lanes are inactive for sv[func]_x